Change adapter to static

maehr opened the pull request:
# Pull request

## Proposed changes

- Change adapter to static to render all pages statically.
- Fix jampack
- Fix several css and html issues
- Improve video controls
- Improve a11y
- Improve webmanifest
- Serve fonts locally
- Serve css locally

<!-- Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue. -->

## Types of changes

<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] New feature (non-breaking change which adds functionality).
- [x] Enhancement (non-breaking change which enhances functionality)
- [x] Bug Fix (non-breaking change which fixes an issue).
- [ ] Breaking change (fix or feature that would cause existing functionality to change).

## Checklist

<!-- Go over all the following points, and put an `x` in all the boxes that apply. If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] I have read the **[README](./README.md)** document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

maehr commented:
The build works, the settings changed due to the switch of the adapter: https://static-site.critique-digitale-ch.pages.dev/

flicksolutions commented:
One general question:
Right now if you preview the build with `npm run build` you will not see the optimized version, because vite somehow builds from _.svelte-kit/output/_ I tried adding ` && jampack .\\.svelte-kit\\output` (windows-specific path just for tests) but then jampack doesn't find the images in the html (because they're in the client folder) and raises issues...
There are three options here:
1. I investigate further and see wether I can get vite to preview from the build folder directly (should work for static pages)
2. live with the errors that jampack generates until a _silent_ option is available
3. live with your preview not being optimized...

How should I proceed?

maehr commented:
The optimization works just fine. You have to run `npm run postbuild` as well (which is done automatically by Cloudflare Pages). The adapter static outputs to the folder `build`. I adjusted `package.json` accordingly.

flicksolutions commented on src/lib/components/Videos.svelte:
```diff
@@ -50,6 +50,10 @@
 	}
+
+	// Automatically hide controls after 2.5 seconds of inactivity
+	$: clearTimeout(showControlsTimeout);
```
this only works for me, if I delete this line... but without the line it works fine...

```suggestion
	
```

flicksolutions replied:
also: add `cursor: pointer;` on line 95.

flicksolutions reviewed (changes requested):
The hiding of the controls doesn't work as proposed. With line 55 deleted, it works.
The proposed changes in package.json are only for DX.

maehr commented:
@flicksolutions I introduced some bugs, please give me one more day before reviewing. Sorry

maehr commented:
> @flicksolutions I introduced some bugs, please give me one more day before reviewing. Sorry

@flicksolutions done! I hope I fixed all requested issues.
